test/snapshots/CodegenMonad.purs

```diff
@@ -6,13 +6,32 @@ import Effect (Effect)
 import Partial.Unsafe (unsafePartial)
 import PureScript.CST.Types (Module)
 import Test.Util (log)
-import Tidy.Codegen (binaryOp, binderVar, declSignature, declValue, exprApp, exprCtor, exprIdent, exprInt, exprOp, exprOpName, printModule, typeApp, typeArrow, typeCtor)
-import Tidy.Codegen.Monad (codegenModule, exporting, importCtor, importFrom, importOp, importOpen, importType, importTypeOp, importValue, write)
+import Tidy.Codegen (binaryOp, binderVar, declSignature, declValue, exprApp, exprIdent, exprInt, exprOp, printModule, typeApp, typeArrow, typeCtor)
+import Tidy.Codegen.Monad (codegenModule, exporting, importCtor, importFrom, importFromAlias, importOp, importOpen, importOpenHiding, importType, importValue, write)
 
 test :: Module Void
 test = unsafePartial do
   codegenModule "Test.Monad" do
     importOpen "Prelude"
+
+    -- Duplicate module import will defer to hidden version
+    importOpen "Prim"
+    importOpenHiding "Prim" $ importType "Type"
+
+    -- Duplicate module import will defer to open version
+    void $ importFrom "Prim.Boolean" $ importType "True"
+    importOpen "Prim.Boolean"
+
+    -- Explicit alias will be used over others
+    void $ importFromAlias "Data.Set" "Set"
+      { val: importValue "NotUsed.isEmpty"
```

Comment:
This says it's not used, so it doesn't get printed, but I'm really curious what actually gets generated 😆